### ISSUE TITLE
use_travis_deploy: Reduce permissions / tips on doing this manually?

### ISSUE THREAD

**Issue by brendan-r** [ropenscilabs/travis#97]
Hello, 

Thank you for your work on this package!

I'm the original author of the [boxr](https://github.com/brendan-r/boxr) package. The users and the maintainer (@ijlyttle) are keen to use the `travis::use_travis_deploy()` to automate the building of a pkgdown site for documentation (and possibly for other things). This is a great idea. However, the permissions required to do this are:

```
This application will be able to read and write all public 
repository data. This includes the following:

    Code
    Issues
    Pull requests
    Wikis
    Settings
    Webhooks and services
    Deploy keys
```

Not that I suspect ropenscilabs of anything nefarious, but I can't wear these.

- Is there a way around granting on-going total access? For example, would a pkgdown site continue to work if I auth the app, run the command, and then de-auth it? I'm assuming that's not the case from the README ("_...Revoking these permissions also invalidates any SSH keys created by this package._"), but I thought I'd check.

- Is there any way I can do these steps 'manually'? If it's useful, I'd be happy to write this up somewhere (take screenshots as I go etc.), if I can get a rough explanation of how to do it.

Sorry to be annoying, and thanks for your package!

Brendan

**Comment by ijlyttle** [comment]
Hi all,

I did some twiddling and some reverse-engineering, I think I can describe the steps that `travis::use_travis_deploy()` takes in terms of how to do it manually.

1. Create a new ssh keypair, perhaps on your laptop using `ssh-keygen`
2. Go to the GitHub repository page - then **Settings** > **Deploy keys**, add the public key there.
3. On your computer, convert the private key using base-64 encoding, copy to clipboard. On my Mac, I used `openssl base64 -A -in <name-of-private-key-file> | pbcopy` (the `-A` prevents newlines from being printed)
4. Go to the Travis repository page - then **More Options** > **Settings**. Add an environment-variable named `id_rsa`. Paste your clipboard-contents into its value. Make sure you are not displaying the value in the log. Save.

At this point, you should have done the job of `travis::use_travis_deploy()`.

**Comment by ijlyttle** [comment]
With the benefit of some sleep, I realize that the key-generating steps could have been produced using some R script, as well as the base-64 encoding step - as you *have already done* with the `use_travis_deploy()` function. 🤦‍♂️

This might be a crazy idea, but when usethis next goes to CRAN, I think there might be enough exposed pieces to support a `manual = TRUE` option to `use_travis_deploy()`, which could:

1. Copy the private key into your clipboard, then direct your browser (or provide a URL) to the Travis repository-settings page, then tell you to create an `id_rsa` environment variable, paste the value, and save.

(hit any key to continue)

2. Copy the public key to your clipboard, then direct your browser (or provide a URL) to the Github settings-keys page, then tell you to create a new key, paste the value, and save.

Or maybe this is a `use_travis_deploy_manual()` function... 

If you think this is worthwhile, I'd be happy to make a PR.

**Comment by seabbs** [comment]
This would be great to have. 

I was doing development in a docker container and couldn't get the automated deployment setup to work. 

I ended up using the keygen etc from `use_travis_deploy` and the steps above (thanks for those :)). 

Thanks, 

Sam

**Comment by pat-s** [comment]
@ijlyttle @brendan-r 
After having merged #115, the auth process is way simpler now. There is no need anymore for authorizing the r-travis app.

When using `travis::use_travis_deploy()` now, 

- a SSH key pair is created 
- the public key is added a Github deploy key to the GH repo
- the private key is added to the Travis repo as a secure env var

Authentication works via Travis API keys.

Let me know if there is still something unclear or not sufficient enough for you to use the {ŧravis} package.
